nexus/reconfigurator/execution/src/datasets.rs

```diff
@@ -269,13 +239,19 @@ mod tests {
     type ControlPlaneTestContext =
         nexus_test_utils::ControlPlaneTestContext<omicron_nexus::Server>;
 
-    fn get_all_datasets_from_zones(
+    fn get_all_crucible_datasets_from_zones(
         blueprint: &Blueprint,
     ) -> Vec<BlueprintDatasetConfig> {
         blueprint
             .all_omicron_zones(BlueprintZoneFilter::All)
             .filter_map(|(_, zone)| {
-                let dataset = zone.zone_type.durable_dataset()?;
+                if !matches!(zone.zone_type, BlueprintZoneType::Crucible(_)) {
+                    return None;
+                }
+                let dataset = zone
```

Comment:
is the type here `CrucibleDataset`?

Reply:
No, it's `DurableDataset`, which contains a reference to an `OmicronZoneDataset`: https://github.com/oxidecomputer/omicron/blob/db656d750a888724b672776fe9b2351c4a683d64/nexus/types/src/deployment/zone_type.rs#L237-L241